lib/fileset/default.nix

```diff
@@ -1012,4 +1013,49 @@ in
             submodules = true;
           }
         );
+
+  /**
+    The empty fileset. It can be useful as a default value or as starting accumulator for a folding operation.
+
+    # Type
+
+    ```
+    empty :: FileSet
+    ```
+  */
+  empty = _emptyWithoutBase;
```

Comment:
I don't see much of a problem with this, considering it's equal to `unions [ ]`, but @infinisil may know of a missing requirement.

Reply:
@infinisil actually had a look at my changes before I created this PR, but it would still be nice to have his opinion on the topic written here.